addons/website_quote/controllers/main.py

```diff
@@ -189,8 +189,11 @@ def payment_transaction_token(self, acquirer_id, order_id, token):
                 'partner_id': Order.partner_id.id,
                 'reference': PaymentTransaction.get_next_reference(Order.name),
                 'sale_order_id': Order.id,
-                'callback_eval': "self.sale_order_id._confirm_online_quote(self)"
+                'callback_ir_model_id': request.env['ir.model'].sudo().search([('model', '=', Order._name)], limit=1).id,
+                'callback_res_id': Order.id,
+                'callback_method': '_confirm_online_quote',
             })
+            Transaction.write({'callback_hash': Transaction._generate_callback_hash()})
```

Comment:
Yes. I think it was done "outside" of the create when experimenting a bit with the code. I will include it in the create: if callback parameters are set, generate the hash.